lib/snippets/deploy-to-gke

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env ruby
+
+# Usage: deploy-to-gke <app's namespace> <gcloud deployment> <gcloud project>
+
+# Requires the following to be available on Shipit machine:
+#  - gcloud binary available on the shipit machine's path
+#  - kubectl binary available in the shipit machine's path
+#  - $GCLOUD_CREDENTIALS_DIR/my-gcloud-project-name.json must exist
+
+# Optionally, the following variables can be used to override script defaults:
+#  - K8S_TEMPLATE_FOLDER: location of Kubernetes files to deploy. Default is config/<environment>/.
+
+require 'open3'
+require 'securerandom'
+require 'erb'
+require 'json'
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+
+class GKEDeployment
+  class FatalDeploymentError < StandardError; end
+
+  def initialize(namespace:, gcloud_deployment:, gcloud_project:, environment:, current_sha:, template_folder: nil)
+    @namespace = namespace
+    @gcloud_deployment = gcloud_deployment
+    @gcloud_project = gcloud_project
+    @current_sha = current_sha
+    enforce_required_parmas
+    @id = current_sha + "-#{SecureRandom.hex(4)}"
```

Comment:
I think we should truncate the sha here. This is going to be appended to pod names to make them unique per deploy, and the max length of a pod name is 63 characters. This would only leave 15 characters for the real pod name. Could be needlessly annoying for our end users.